packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java

```diff
@@ -1414,14 +1439,54 @@ private int registerWifiNetworkDeprecated(android.net.wifi.WifiConfiguration con
                 || wifiConfig.BSSID.equals(conf.BSSID))) {
           conf.networkId = wifiConfig.networkId;
           registeredNetwork = wifiConfig.networkId;
-          updateNetwork = moWiFi.updateNetwork(conf);
+          //only try to update the configuration if joinOnce is false
+          //otherwise use the new add/update method
+          if (joinOnce != null && !joinOnce.booleanValue()) {
+            updateNetwork = moWiFi.updateNetwork(conf);
+            //required for pre API 26
+            moWiFi.saveConfiguration();
+          }
+          //Android 6.0 and higher no longer allows you to update a network that wasn't created
+          //from our app, nor delete it, nor add a network with the same SSID
+          //See https://developer.android.com/about/versions/marshmallow/android-6.0-changes.html#behavior-network
+          if (updateNetwork == -1) {
+            //we add some random number to the conf SSID, add that network, then change the SSID
+            // back to circumvent the issue
+            String ssid = conf.SSID;
+            Random random = new Random(System.currentTimeMillis());
+            //loop in the rare case that the generated ssid already exists
+            for (int i = 0; i < 20; i++) {
+              int randomInteger = random.nextInt(10000);
+              //create a valid SSID with max length of 32
+              String ssidRandomized = ssid + randomInteger;
```

Comment:
`ssid` will be wrapped around quote (`"`) - that should be trimmed, right?

Reply:
It could be, but it isn't necessary, since we mostly care about the length anyway. It would be "cleaner" to trim it, yes.